chiselFrontend/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -186,6 +187,16 @@ private[chisel3] object Builder {
   def whenDepth_=(target: Int): Unit = {
     dynamicContext.whenDepth = target
   }
+  def currentClockAndReset: Option[ClockAndReset] = dynamicContext.currentClockAndReset
+  def currentClockAndReset_=(target: Option[ClockAndReset]): Unit = {
+    dynamicContext.currentClockAndReset = target
+  }
+  def forcedClockAndReset: ClockAndReset = currentClockAndReset match {
+    case Some(clockAndReset) => clockAndReset
+    case None => throwException("Error: No implicit clock and reset.")
+  }
+  def forcedClock: Clock = forcedClockAndReset.clock
+  def forcedReset: Bool = forcedClockAndReset.reset
```

Comment:
What's the idea behind the "forced" terminology, instead of "current" or "implicit"?